From 22c24ba0a7d1b5514bfffe3ba22850d8f0389586 Mon Sep 17 00:00:00 2001
From: hienhuynh <hien.huynh.px@renesas.com>
Date: Wed, 24 Jun 2020 13:00:12 +0700
Subject: [PATCH 6/6] renesas: rzg: add support Silinux EK874 rev3 board

Silinux EK874 rev3 board uses G2E LSI v1.1.
G2E LSI v1.0 is used for Silinux EK874 rev2 and rev1 board.
Compare the different LSI to identify the rev of board.

Signed-off-by: hienhuynh <hien.huynh.px@renesas.com>
---
 drivers/renesas/rzg/board/board.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/drivers/renesas/rzg/board/board.c b/drivers/renesas/rzg/board/board.c
index f933269..b2f2748 100644
--- a/drivers/renesas/rzg/board/board.c
+++ b/drivers/renesas/rzg/board/board.c
@@ -150,8 +150,12 @@ get_type:
 	boardInfo = mmio_read_32(GPIO_INDT5) & (GP5_19_BIT |GP5_21_BIT);
 	*rev = (((boardInfo & GP5_19_BIT) >> 14) | ((boardInfo & GP5_21_BIT) >> 17)) + 0x30;
 #else
-	read_rev = (uint8_t)(board_id & BOARD_REV_MASK);
-	*rev = board_tbl[*type][read_rev];
+	if (mmio_read_32(RCAR_PRR) & RCAR_MINOR_MASK)
+		*rev = 0x30U;
+	else {
+		read_rev = (uint8_t)(board_id & BOARD_REV_MASK);
+		*rev = board_tbl[*type][read_rev];
+	}
 #endif
 	return ret;
 }
-- 
2.7.4

